api/envoy/config/core/v3/base.proto

```diff
@@ -308,7 +308,7 @@ message KeyValue {
   string key = 1 [(validate.rules).string = {min_len: 1 max_bytes: 16384}];
 
   // The value of the key/value pair.
-  bytes value = 2;
+  string value = 2;
```

Comment:
Please no. `bytes` is used here by design.

Reply:
doh, left an unsent github comment. I was wondering about the use of `bytes`? IIUC this makes command substitution a no-go which is the primary use-case I have in mind (appending a query param from metadata). It also is pretty inconsistent - elsewhere I see an option for a `value` string and a `raw_value` bytes.

Reply:
> IIUC this makes command substitution a no-go which is the primary use-case I have in mind (appending a query param from metadata).

The `bytes` is used to support non-UTF8. But I think it's also OK to use it as a command formatter for Envoy. But it may not be Okay for non-C++ xDS clients.

@htuch WDYT? should we use string value as value with command formatter and use bytes value as raw value?

Reply:
It's same as the `HeaderValue` message. The `value` field may be a string with substitution formatter like `%REQ(CUSTOM-KEY)%`. `bytes` should be fine for C++ xDS client like Envoy in this scenario, because `bytes` and string has no essential difference. But not sure if it's OK for non-C++ client.

cc @htuch

Reply:
@markdroth